Installation issue: `expect`

### Steps to reproduce the issue

```console
$ spack spec expect%gcc
Input spec
--------------------------------
 -   expect%gcc

Concretized
--------------------------------
 -   expect@5.45.4%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]      ^autoconf@2.72%clang@18.1.3 build_system=autotools arch=linux-ubuntu24.04-aarch64
[e]          ^glibc@2.39%clang@18.1.3 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]          ^perl@5.40.0%clang@18.1.3+cpanm+opcode+open+shared+threads build_system=generic arch=linux-ubuntu24.04-aarch64
[+]              ^berkeley-db@18.1.40%clang@18.1.3+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu24.04-aarch64
[+]              ^bzip2@1.0.8%clang@18.1.3~debug~pic+shared build_system=generic arch=linux-ubuntu24.04-aarch64
[+]              ^gdbm@1.23%clang@18.1.3 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]                  ^readline@8.2%clang@18.1.3 build_system=autotools patches=bbf97f1 arch=linux-ubuntu24.04-aarch64
[+]                      ^ncurses@6.5%clang@18.1.3~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-ubuntu24.04-aarch64
[+]                          ^pkgconf@2.2.0%clang@18.1.3 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]      ^automake@1.16.5%clang@18.1.3 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]      ^gcc-runtime@14.2.0%gcc@14.2.0 build_system=generic arch=linux-ubuntu24.04-aarch64
[e]      ^glibc@2.39%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]      ^gmake@4.4.1%clang@18.1.3~guile build_system=generic arch=linux-ubuntu24.04-aarch64
[+]      ^gnuconfig@2022-09-17%clang@18.1.3 build_system=generic arch=linux-ubuntu24.04-aarch64
[+]      ^libtool@2.4.7%clang@18.1.3 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]          ^findutils@4.9.0%clang@18.1.3 build_system=autotools patches=440b954 arch=linux-ubuntu24.04-aarch64
[+]      ^m4@1.4.19%clang@18.1.3+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu24.04-aarch64
[+]          ^diffutils@3.10%clang@18.1.3 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]              ^libiconv@1.17%clang@18.1.3 build_system=autotools libs=shared,static arch=linux-ubuntu24.04-aarch64
[+]          ^libsigsegv@2.14%clang@18.1.3 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]      ^tcl@8.6.12%gcc@14.2.0 build_system=autotools arch=linux-ubuntu24.04-aarch64
[+]          ^zlib-ng@2.2.1%clang@18.1.3+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu24.04-aarch64
```


### Error message

<details><summary>Error message</summary>

<pre>
==> expect: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/tmp/yizeyi18/spack-stage/spack-stage-expect-5.45.4-djiqup5ycsx4kgx2rdyzdqu6upa24tnw/spack-src/configure' '--prefix=/home/yizeyi18/soft/spack/opt/spack/linux-ubuntu24.04-aarch64/gcc-14.2.0/expect-5.45.4-djiqup5ycsx4kgx2rdyzdqu6upa24tnw' '--exec-prefix=/home/yizeyi18/soft/spack/opt/spack/linux-ubuntu24.04-aarch64/gcc-14.2.0/expect-5.45.4-djiqup5ycsx4kgx2rdyzdqu6upa24tnw' '--enable-threads' '--enable-shared' '--enable-64bit' '--with-tcl=/home/yizeyi18/soft/spack/opt/spack/linux-ubuntu24.04-aarch64/gcc-14.2.0/tcl-8.6.12-m2rf2hjhvocg432df7zwm7alsmkjlqdi/lib' '--with-tclinclude=/home/yizeyi18/soft/spack/opt/spack/linux-ubuntu24.04-aarch64/gcc-14.2.0/tcl-8.6.12-m2rf2hjhvocg432df7zwm7alsmkjlqdi/include'

192 errors found in build log:
     18     autoreconf: configure.in: tracing
     19     autoreconf: configure.in: not using Libtool
     20     autoreconf: configure.in: not using Intltool
     21     autoreconf: configure.in: not using Gtkdoc
     22     autoreconf: running: /home/yizeyi18/soft/spack/opt/spack/linux-ubuntu24.04-aarch64/clang-18.1.3/autoconf-2.
            72-wjgqzyjob7duec75u5iz4sly2qwfwgof/bin/autoconf --include=/home/yizeyi18/soft/spack/opt/spack/linux-ubuntu
            24.04-aarch64/clang-18.1.3/libtool-2.4.7-4r6scbi7rgv2c64ehty7ufebw4e6npce/share/aclocal --force
     23     configure.in:8: warning: The macro 'AC_HELP_STRING' is obsolete.
  >> 24     configure.in:8: You should run autoupdate.
  >> 25     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:204: AC_HELP_STRING is expanded from...
  >> 26     ../tclconfig/tcl.m4:45: TEA_PATH_TCLCONFIG is expanded from...
  >> 27     configure.in:8: the top level
     28     configure.in:11: warning: The macro 'AC_TRY_COMPILE' is obsolete.
  >> 29     configure.in:11: You should run autoupdate.
  >> 30     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2845: AC_TRY_COMPILE is expanded from...
  >> 31     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 32     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 33     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 34     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2270: AC_CACHE_CHECK is expanded from...
  >> 35     ../tclconfig/tcl.m4:3121: TEA_SETUP_COMPILER is expanded from...
  >> 36     configure.in:11: the top level
     37     configure.in:11: warning: The macro 'AC_TRY_LINK' is obsolete.
  >> 38     configure.in:11: You should run autoupdate.
  >> 39     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2918: AC_TRY_LINK is expanded from...
  >> 40     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 41     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 42     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 43     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2270: AC_CACHE_CHECK is expanded from...
  >> 44     ../tclconfig/tcl.m4:2159: TEA_MISSING_POSIX_HEADERS is expanded from...
  >> 45     ../tclconfig/tcl.m4:3121: TEA_SETUP_COMPILER is expanded from...
  >> 46     configure.in:11: the top level
     47     configure.in:11: warning: The macro 'AC_HAVE_HEADERS' is obsolete.
  >> 48     configure.in:11: You should run autoupdate.
  >> 49     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /oldnames.m4:35: AC_HAVE_HEADERS is expanded from...
  >> 50     ../tclconfig/tcl.m4:2159: TEA_MISSING_POSIX_HEADERS is expanded from...
  >> 51     ../tclconfig/tcl.m4:3121: TEA_SETUP_COMPILER is expanded from...
  >> 52     configure.in:11: the top level
     53     configure.in:14: warning: AC_OUTPUT should be used without arguments.
  >> 54     configure.in:14: You should run autoupdate.
     55     configure.in:14: warning: AC_C_BIGENDIAN should be used with AC_CONFIG_HEADERS
     56     autoreconf: configure.in: not using Autoheader
     57     autoreconf: configure.in: not using Automake
     58     autoreconf: '../tclconfig/install-sh' is updated
     59     autoreconf: Leaving directory 'testsuite'
     60     autoreconf: configure.in: not using Libtool
     61     autoreconf: configure.in: not using Intltool
     62     autoreconf: configure.in: not using Gtkdoc
     63     autoreconf: running: /home/yizeyi18/soft/spack/opt/spack/linux-ubuntu24.04-aarch64/clang-18.1.3/autoconf-2.
            72-wjgqzyjob7duec75u5iz4sly2qwfwgof/bin/autoconf --include=/home/yizeyi18/soft/spack/opt/spack/linux-ubuntu
            24.04-aarch64/clang-18.1.3/libtool-2.4.7-4r6scbi7rgv2c64ehty7ufebw4e6npce/share/aclocal --force
     64     configure.in:21: warning: The macro 'AC_HELP_STRING' is obsolete.
  >> 65     configure.in:21: You should run autoupdate.
  >> 66     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:204: AC_HELP_STRING is expanded from...
  >> 67     tclconfig/tcl.m4:45: TEA_PATH_TCLCONFIG is expanded from...
  >> 68     configure.in:21: the top level
     69     configure.in:46: warning: The macro 'AC_TRY_COMPILE' is obsolete.
  >> 70     configure.in:46: You should run autoupdate.
  >> 71     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2845: AC_TRY_COMPILE is expanded from...
  >> 72     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 73     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 74     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 75     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2270: AC_CACHE_CHECK is expanded from...
  >> 76     tclconfig/tcl.m4:3121: TEA_SETUP_COMPILER is expanded from...
  >> 77     configure.in:46: the top level
     78     configure.in:46: warning: The macro 'AC_TRY_LINK' is obsolete.
  >> 79     configure.in:46: You should run autoupdate.
  >> 80     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2918: AC_TRY_LINK is expanded from...
  >> 81     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 82     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 83     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 84     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2270: AC_CACHE_CHECK is expanded from...
  >> 85     tclconfig/tcl.m4:2159: TEA_MISSING_POSIX_HEADERS is expanded from...
  >> 86     tclconfig/tcl.m4:3121: TEA_SETUP_COMPILER is expanded from...
  >> 87     configure.in:46: the top level
     88     configure.in:46: warning: The macro 'AC_HAVE_HEADERS' is obsolete.
  >> 89     configure.in:46: You should run autoupdate.
  >> 90     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /oldnames.m4:35: AC_HAVE_HEADERS is expanded from...
  >> 91     tclconfig/tcl.m4:2159: TEA_MISSING_POSIX_HEADERS is expanded from...
  >> 92     tclconfig/tcl.m4:3121: TEA_SETUP_COMPILER is expanded from...
  >> 93     configure.in:46: the top level
     94     configure.in:80: warning: The macro 'AC_HELP_STRING' is obsolete.
  >> 95     configure.in:80: You should run autoupdate.
  >> 96     /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:204: AC_HELP_STRING is expanded from...
  >> 97     tclconfig/tcl.m4:738: TEA_ENABLE_THREADS is expanded from...
  >> 98     configure.in:80: the top level
     99     configure.in:87: warning: The macro 'AC_HELP_STRING' is obsolete.
  >> 100    configure.in:87: You should run autoupdate.
  >> 101    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:204: AC_HELP_STRING is expanded from...
  >> 102    tclconfig/tcl.m4:680: TEA_ENABLE_SHARED is expanded from...
  >> 103    configure.in:87: the top level
     104    configure.in:95: warning: The macro 'AC_HELP_STRING' is obsolete.
  >> 105    configure.in:95: You should run autoupdate.
  >> 106    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:204: AC_HELP_STRING is expanded from...
  >> 107    tclconfig/tcl.m4:1051: TEA_CONFIG_CFLAGS is expanded from...
  >> 108    configure.in:95: the top level
     109    configure.in:95: warning: The macro 'AC_TRY_LINK' is obsolete.
  >> 110    configure.in:95: You should run autoupdate.
  >> 111    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2918: AC_TRY_LINK is expanded from...
  >> 112    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 113    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 114    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 115    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2270: AC_CACHE_CHECK is expanded from...
  >> 116    tclconfig/tcl.m4:1051: TEA_CONFIG_CFLAGS is expanded from...
  >> 117    configure.in:95: the top level
     118    configure.in:95: warning: The macro 'AC_HELP_STRING' is obsolete.
  >> 119    configure.in:95: You should run autoupdate.
  >> 120    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:204: AC_HELP_STRING is expanded from...
  >> 121    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 122    tclconfig/tcl.m4:1051: TEA_CONFIG_CFLAGS is expanded from...
  >> 123    configure.in:95: the top level
     124    configure.in:95: warning: The macro 'AC_TRY_LINK' is obsolete.
  >> 125    configure.in:95: You should run autoupdate.
  >> 126    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2918: AC_TRY_LINK is expanded from...
  >> 127    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 128    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 129    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 130    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2270: AC_CACHE_CHECK is expanded from...
  >> 131    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 132    tclconfig/tcl.m4:1051: TEA_CONFIG_CFLAGS is expanded from...
  >> 133    configure.in:95: the top level
     134    configure.in:95: warning: The macro 'AC_TRY_LINK' is obsolete.
  >> 135    configure.in:95: You should run autoupdate.
  >> 136    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2918: AC_TRY_LINK is expanded from...
  >> 137    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 138    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 139    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 140    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2270: AC_CACHE_CHECK is expanded from...
  >> 141    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 142    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 143    tclconfig/tcl.m4:1051: TEA_CONFIG_CFLAGS is expanded from...
  >> 144    configure.in:95: the top level
     145    configure.in:95: warning: The macro 'AC_TRY_COMPILE' is obsolete.
  >> 146    configure.in:95: You should run autoupdate.
  >> 147    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2845: AC_TRY_COMPILE is expanded from...
  >> 148    tclconfig/tcl.m4:2581: TEA_TCL_EARLY_FLAG is expanded from...
  >> 149    tclconfig/tcl.m4:2594: TEA_TCL_EARLY_FLAGS is expanded from...
  >> 150    tclconfig/tcl.m4:1051: TEA_CONFIG_CFLAGS is expanded from...
  >> 151    configure.in:95: the top level
     152    configure.in:95: warning: The macro 'AC_TRY_COMPILE' is obsolete.
  >> 153    configure.in:95: You should run autoupdate.
  >> 154    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2845: AC_TRY_COMPILE is expanded from...
  >> 155    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 156    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 157    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 158    tclconfig/tcl.m4:2629: TEA_TCL_64BIT_FLAGS is expanded from...
  >> 159    tclconfig/tcl.m4:1051: TEA_CONFIG_CFLAGS is expanded from...
  >> 160    configure.in:95: the top level
     161    configure.in:95: warning: The macro 'AC_TRY_COMPILE' is obsolete.
  >> 162    configure.in:95: You should run autoupdate.
  >> 163    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2845: AC_TRY_COMPILE is expanded from...
  >> 164    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 165    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 166    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 167    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2270: AC_CACHE_CHECK is expanded from...
  >> 168    tclconfig/tcl.m4:2629: TEA_TCL_64BIT_FLAGS is expanded from...
  >> 169    tclconfig/tcl.m4:1051: TEA_CONFIG_CFLAGS is expanded from...
  >> 170    configure.in:95: the top level
     171    configure.in:101: warning: The macro 'AC_HELP_STRING' is obsolete.
  >> 172    configure.in:101: You should run autoupdate.
  >> 173    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:204: AC_HELP_STRING is expanded from...
  >> 174    tclconfig/tcl.m4:867: TEA_ENABLE_SYMBOLS is expanded from...
  >> 175    configure.in:101: the top level
     176    configure.in:141: warning: The macro 'AC_HAVE_HEADERS' is obsolete.
  >> 177    configure.in:141: You should run autoupdate.
  >> 178    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /oldnames.m4:35: AC_HAVE_HEADERS is expanded from...
  >> 179    configure.in:141: the top level
     180    configure.in:150: warning: The macro 'AC_TYPE_SIGNAL' is obsolete.
  >> 181    configure.in:150: You should run autoupdate.
  >> 182    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /types.m4:805: AC_TYPE_SIGNAL is expanded from...
  >> 183    configure.in:150: the top level
     184    configure.in:156: warning: The macro 'AC_HEADER_TIME' is obsolete.
  >> 185    configure.in:156: You should run autoupdate.
  >> 186    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /headers.m4:702: AC_HEADER_TIME is expanded from...
  >> 187    tclconfig/tcl.m4:2374: TEA_TIME_HANDLER is expanded from...
  >> 188    configure.in:156: the top level
     189    configure.in:156: warning: The macro 'AC_TRY_COMPILE' is obsolete.
  >> 190    configure.in:156: You should run autoupdate.
  >> 191    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2845: AC_TRY_COMPILE is expanded from...
  >> 192    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:690: _AS_IF_ELSE is expanded from...
  >> 193    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/m4sugar/
            m4sh.m4:697: AS_IF is expanded from...
  >> 194    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2249: AC_CACHE_VAL is expanded from...
  >> 195    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2270: AC_CACHE_CHECK is expanded from...
  >> 196    tclconfig/tcl.m4:2374: TEA_TIME_HANDLER is expanded from...
  >> 197    configure.in:156: the top level
     198    configure.in:168: warning: The macro 'AC_TRY_LINK' is obsolete.
  >> 199    configure.in:168: You should run autoupdate.
  >> 200    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2918: AC_TRY_LINK is expanded from...
  >> 201    configure.in:168: the top level
     202    configure.in:254: warning: The macro 'AC_CANONICAL_SYSTEM' is obsolete.
  >> 203    configure.in:254: You should run autoupdate.
  >> 204    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2081: AC_CANONICAL_SYSTEM is expanded from...
  >> 205    configure.in:254: the top level
     206    configure.in:455: warning: The macro 'AC_TRY_LINK' is obsolete.
  >> 207    configure.in:455: You should run autoupdate.
  >> 208    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2918: AC_TRY_LINK is expanded from...
  >> 209    configure.in:455: the top level
     210    configure.in:470: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 211    configure.in:470: You should run autoupdate.
  >> 212    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 213    configure.in:470: the top level
     214    configure.in:489: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 215    configure.in:489: You should run autoupdate.
  >> 216    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 217    configure.in:489: the top level
     218    configure.in:537: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 219    configure.in:537: You should run autoupdate.
  >> 220    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 221    configure.in:537: the top level
     222    configure.in:642: warning: The macro 'AC_HEADER_EGREP' is obsolete.
  >> 223    configure.in:642: You should run autoupdate.
  >> 224    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /oldnames.m4:37: AC_HEADER_EGREP is expanded from...
  >> 225    configure.in:642: the top level
     226    configure.in:715: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 227    configure.in:715: You should run autoupdate.
  >> 228    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 229    configure.in:715: the top level
     230    configure.in:740: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 231    configure.in:740: You should run autoupdate.
  >> 232    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 233    configure.in:740: the top level
     234    configure.in:757: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 235    configure.in:757: You should run autoupdate.
  >> 236    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 237    configure.in:757: the top level
     238    configure.in:779: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 239    configure.in:779: You should run autoupdate.
  >> 240    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 241    configure.in:779: the top level
     242    configure.in:801: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 243    configure.in:801: You should run autoupdate.
  >> 244    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 245    configure.in:801: the top level
     246    configure.in:825: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 247    configure.in:825: You should run autoupdate.
  >> 248    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 249    configure.in:825: the top level
     250    configure.in:868: warning: The macro 'AC_HAVE_FUNCS' is obsolete.
  >> 251    configure.in:868: You should run autoupdate.
  >> 252    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /functions.m4:230: AC_HAVE_FUNCS is expanded from...
  >> 253    configure.in:868: the top level
     254    configure.in:869: warning: The macro 'AC_HAVE_FUNCS' is obsolete.
  >> 255    configure.in:869: You should run autoupdate.
  >> 256    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /functions.m4:230: AC_HAVE_FUNCS is expanded from...
  >> 257    configure.in:869: the top level
     258    configure.in:878: warning: The macro 'AC_TRY_RUN' is obsolete.
  >> 259    configure.in:878: You should run autoupdate.
  >> 260    /tmp/yizeyi18/spack-stage/spack-stage-autoconf-2.72-wjgqzyjob7duec75u5iz4sly2qwfwgof/spack-src/lib/autoconf
            /general.m4:2995: AC_TRY_RUN is expanded from...
  >> 261    configure.in:878: the top level
     262    configure.in:1059: warning: AC_OUTPUT should be used without arguments.
  >> 263    configure.in:1059: You should run autoupdate.
     264    configure.in:1059: warning: AC_C_BIGENDIAN should be used with AC_CONFIG_HEADERS
     265    autoreconf: configure.in: not using Autoheader
     266    autoreconf: configure.in: not using Automake
     267    autoreconf: 'tclconfig/config.sub' is updated
     268    autoreconf: 'tclconfig/config.guess' is updated
     269    autoreconf: Leaving directory '.'

     ...

     420    checking event handling... via select
     421    checking for _getpty... no
     422    checking for getpty... no
     423    checking whether setpgrp requires zero arguments... yes
     424    checking for SV-style timezone... no
     425    checking for long file names... yes
  >> 426    configure: error: could not find source file 'pty_.c'

See build log for details:
  /tmp/yizeyi18/spack-stage/spack-stage-expect-5.45.4-djiqup5ycsx4kgx2rdyzdqu6upa24tnw/spack-build-out.txt
</pre></details>


### Information on your system

* **Spack:** 0.23.0.dev0 (c99518709a65bea10d96017a4c06c55774218d6e)
* **Python:** 3.12.3
* **Platform:** linux-ubuntu24.04-aarch64

### Additional information

The error comes from restrict error on implicit return type and restrict include check of `stdlib.h` since gcc-13 and clang-15, which had been patched by expect community: https://sourceforge.net/p/expect/patches/24/ 

Also the spack community had gived a partly fix in spack/spack#21232 .

[spack-build-env.txt](https://github.com/user-attachments/files/17349200/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/17349202/spack-build-out.txt)
[config.log](https://github.com/user-attachments/files/17349203/config.log)
[expect-configure-c99.patch](https://github.com/user-attachments/files/17349204/expect-configure-c99.patch)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/expect/func_proto.patch
+++ b/repos/spack_repo/builtin/packages/expect/func_proto.patch
@@ -1,0 +1,333 @@
+From c5e79881646e792f4f686ff4de5de14b1d1a18d2 Mon Sep 17 00:00:00 2001
+From: Khem Raj <raj.khem@gmail.com>
+Date: Mon, 15 Aug 2022 18:25:23 -0700
+Subject: [PATCH] Add prototype to function definitions
+
+Compilers like clang has started erroring out on implicit-function-declaration
+therefore arrange the relevant include files where needed.
+
+---
+ exp_chan.c     | 5 +++--
+ exp_clib.c     | 4 +++-
+ exp_main_sub.c | 5 +++++
+ pty_termios.c  | 4 ++++
+ 4 files changed, 15 insertions(+), 3 deletions(-)
+
+diff --git a/exp_chan.c b/exp_chan.c
+index 79f486c..50375d3 100644
+--- a/exp_chan.c
++++ b/exp_chan.c
+@@ -35,6 +35,7 @@
+ #include "exp_prog.h"
+ #include "exp_command.h"
+ #include "exp_log.h"
++#include "exp_event.h" /* exp_background_channelhandler */
+ #include "tcldbg.h" /* Dbg_StdinMode */
+ 
+ extern int		expSetBlockModeProc _ANSI_ARGS_((int fd, int mode));
+@@ -631,7 +632,7 @@ expWaitOnOne() {
+ }
+ 
+ void
+-exp_background_channelhandlers_run_all()
++exp_background_channelhandlers_run_all(void)
+ {
+     ThreadSpecificData *tsdPtr = TCL_TSD_INIT(&dataKey);
+     ExpState *esPtr;
+@@ -760,7 +761,7 @@ expCreateChannel(interp,fdin,fdout,pid)
+ }
+ 
+ void
+-expChannelInit() {
++expChannelInit(void) {
+     ThreadSpecificData *tsdPtr = TCL_TSD_INIT(&dataKey);
+ 
+     tsdPtr->channelCount = 0;
+diff --git a/exp_clib.c b/exp_clib.c
+index b21fb5d..8f31fc3 100644
+--- a/exp_clib.c
++++ b/exp_clib.c
+@@ -9,13 +9,14 @@ would appreciate credit if this program or parts of it are used.
+ 
+ #include "expect_cf.h"
+ #include <stdio.h>
++#include <unistd.h>
+ #include <setjmp.h>
+ #ifdef HAVE_INTTYPES_H
+ #  include <inttypes.h>
+ #endif
+ #include <sys/types.h>
+ #include <sys/ioctl.h>
+-
++#include <sys/wait.h>
+ #ifdef TIME_WITH_SYS_TIME
+ # include <sys/time.h>
+ # include <time.h>
+@@ -1738,6 +1739,7 @@ int exp_getptyslave();
+ #define sysreturn(x)	return(errno = x, -1)
+ 
+ void exp_init_pty();
++void exp_init_tty();
+ 
+ /*
+    The following functions are linked from the Tcl library.  They
+diff --git a/exp_main_sub.c b/exp_main_sub.c
+index bf6c4be..f53b89e 100644
+--- a/exp_main_sub.c
++++ b/exp_main_sub.c
+@@ -61,6 +61,11 @@ int exp_cmdlinecmds = FALSE;
+ int exp_interactive =  FALSE;
+ int exp_buffer_command_input = FALSE;/* read in entire cmdfile at once */
+ int exp_fgets();
++int exp_tty_cooked_echo(
++    Tcl_Interp *interp,
++    exp_tty *tty_old,
++    int *was_raw,
++    int *was_echo);
+ 
+ Tcl_Interp *exp_interp;	/* for use by signal handlers who can't figure out */
+ 			/* the interpreter directly */
+diff --git a/pty_termios.c b/pty_termios.c
+index c605b23..80ed5e7 100644
+--- a/pty_termios.c
++++ b/pty_termios.c
+@@ -7,6 +7,7 @@ would appreciate credit if you use this file or parts of it.
+ 
+ */
+ 
++#include <pty.h> /* openpty */
+ #include <stdio.h>
+ #include <signal.h>
+ 
+@@ -15,6 +16,9 @@ would appreciate credit if you use this file or parts of it.
+ #endif
+ 
+ #include "expect_cf.h"
++#include "tclInt.h"
++
++extern char * expErrnoMsg    _ANSI_ARGS_((int));
+ 
+ /*
+    The following functions are linked from the Tcl library.  They
+-- 
+2.37.2
+
+diff --git a/exp_chan.c b/exp_chan.c
+index c92e26b6fbd02305..944200a63b102672 100644
+--- a/exp_chan.c
++++ b/exp_chan.c
+@@ -60,7 +60,7 @@ void			exp_background_channelhandler _ANSI_ARGS_((ClientData,
+ 
+ Tcl_ChannelType expChannelType = {
+     "exp",				/* Type name. */
+-    ExpBlockModeProc,			/* Set blocking/nonblocking mode.*/
++    TCL_CHANNEL_VERSION_2,
+     ExpCloseProc,			/* Close proc. */
+     ExpInputProc,			/* Input proc. */
+     ExpOutputProc,			/* Output proc. */
+@@ -70,6 +70,7 @@ Tcl_ChannelType expChannelType = {
+     ExpWatchProc,			/* Initialize notifier. */
+     ExpGetHandleProc,			/* Get OS handles out of channel. */
+     NULL,				/* Close2 proc */
++    ExpBlockModeProc,			/* Set blocking/nonblocking mode.*/
+ };
+ 
+ typedef struct ThreadSpecificData {
+Avoid calling exit without declaring the function.
+
+Add missing <string.h> include for memcpy.
+
+Use AC_TYPE_SIGNAL to fix REARM_SIG check.  Add missing includes.
+
+Fix various implicit int return types of main.
+
+diff --git a/configure.in b/configure.in
+index 51558fa14d2bcf7e..055c88fbd8797eaa 100755
+--- a/configure.in
++++ b/configure.in
+@@ -452,7 +452,11 @@ AC_CHECK_FUNC(siglongjmp, AC_DEFINE(HAVE_SIGLONGJMP))
+ # because Unixware 2.0 handles it specially and refuses to compile
+ # autoconf's automatic test that is a call with no arguments
+ AC_MSG_CHECKING([for memcpy])
+-AC_TRY_LINK(,[
++AC_TRY_LINK([
++#ifdef HAVE_STRING_H
++#include <string.h>
++#endif
++],[
+ char *s1, *s2;
+ memcpy(s1,s2,0);
+ ],
+@@ -469,6 +473,7 @@ memcpy(s1,s2,0);
+ AC_MSG_CHECKING([if WNOHANG requires _POSIX_SOURCE])
+ AC_TRY_RUN([
+ #include <sys/wait.h>
++int
+ main() {
+ #ifndef WNOHANG
+ 	return 0;
+@@ -489,6 +494,7 @@ rm -rf wnohang
+ AC_TRY_RUN([
+ #include <stdio.h>
+ #include <sys/wait.h>
++int
+ main() {
+ #ifdef WNOHANG
+ 	FILE *fp = fopen("wnohang","w");
+@@ -527,16 +533,21 @@ else
+   AC_DEFINE(SELECT_MASK_TYPE, fd_set)
+ fi
+ 
+-dnl # Check for the data type of the function used in signal(). This
+-dnl # must be before the test for rearming.
+-dnl # echo checking return type of signal handlers
+-dnl AC_HEADER_EGREP([(void|sighandler_t).*signal], signal.h, retsigtype=void,AC_DEFINE(RETSIGTYPE, int) retsigtype=int)
++AC_TYPE_SIGNAL
+ 
+ # FIXME: check if alarm exists
+ AC_MSG_CHECKING([if signals need to be re-armed])
+ AC_TRY_RUN([
+ #include <signal.h>
+-#define RETSIGTYPE $retsigtype
++#ifdef HAVE_STDLIB_H
++# include <stdlib.h>
++#endif
++#ifdef HAVE_UNISTD_H
++# include <unistd.h>
++#endif
++#ifndef NO_SYS_WAIT_H
++# include <sys/wait.h>
++#endif
+ 
+ int signal_rearms = 0;
+ 
+@@ -553,6 +564,7 @@ int n;
+ signal_rearms++;
+ }
+ 
++int
+ main()
+ {
+ 	signal(SIGINT,parent_sigint_handler);
+@@ -714,10 +726,11 @@ fi
+ AC_MSG_CHECKING([for struct sgttyb])
+ AC_TRY_RUN([
+ #include <sgtty.h>
++int
+ main()
+ {
+   struct sgttyb tmp;
+-  exit(0);
++  return 0;
+ }],
+         AC_MSG_RESULT(yes)
+         AC_DEFINE(HAVE_SGTTYB)
+@@ -738,10 +751,11 @@ if test $mach -eq 0 ; then
+   # pty_termios.c is set up to handle pty_termio.
+   AC_MSG_CHECKING([for struct termio])
+   AC_TRY_RUN([#include <termio.h>
++  int
+   main()
+   {
+     struct termio tmp;
+-    exit(0);
++    return 0;
+   }],
+         AC_DEFINE(HAVE_TERMIO)
+         PTY_TYPE=termios
+@@ -760,10 +774,11 @@ if test $mach -eq 0 ; then
+ #  include <inttypes.h>
+ #  endif
+ #  include <termios.h>
++  int
+   main()
+   {
+     struct termios tmp;
+-    exit(0);
++    return 0;
+   }],
+         AC_DEFINE(HAVE_TERMIOS)
+         PTY_TYPE=termios
+@@ -782,6 +797,7 @@ AC_TRY_RUN([
+ #include <inttypes.h>
+ #endif
+ #include <termios.h>
++int
+ main() {
+ #if defined(TCGETS) || defined(TCGETA)
+ 	return 0;
+@@ -804,6 +820,7 @@ AC_TRY_RUN([
+ #include <inttypes.h>
+ #endif
+ #include <termios.h>
++int
+ main() {
+ #ifdef TIOCGWINSZ
+ 	return 0;
+@@ -823,6 +840,7 @@ main() {
+ AC_MSG_CHECKING([for Cray-style ptys])
+ SETUID=":"
+ AC_TRY_RUN([
++int
+ main(){
+ #ifdef CRAY
+ 	return 0;
+@@ -878,12 +896,13 @@ AC_MSG_CHECKING([for SV-style timezone])
+ AC_TRY_RUN([
+ extern char *tzname[2];
+ extern int daylight;
++int
+ main()
+ {
+   int *x = &daylight;
+   char **y = tzname;
+ 
+-  exit(0);
++  return 0;
+ }],
+ 	AC_DEFINE(HAVE_SV_TIMEZONE)
+ 	AC_MSG_RESULT(yes),
+diff --git a/tclconfig/tcl.m4 b/tclconfig/tcl.m4
+index 0689cab3da994068..ebe839e5553ba520 100644
+--- a/tclconfig/tcl.m4
++++ b/tclconfig/tcl.m4
+@@ -2400,7 +2400,7 @@ AC_DEFUN([TEA_TIME_HANDLER], [
+ 	AC_TRY_COMPILE([#include <time.h>],
+ 	    [extern long timezone;
+ 	    timezone += 1;
+-	    exit (0);],
++	    return 0;],
+ 	    tcl_cv_timezone_long=yes, tcl_cv_timezone_long=no)])
+     if test $tcl_cv_timezone_long = yes ; then
+ 	AC_DEFINE(HAVE_TIMEZONE_VAR, 1, [Should we use the global timezone variable?])
+@@ -2412,7 +2412,7 @@ AC_DEFUN([TEA_TIME_HANDLER], [
+ 	    AC_TRY_COMPILE([#include <time.h>],
+ 		[extern time_t timezone;
+ 		timezone += 1;
+-		exit (0);],
++		return 0;],
+ 		tcl_cv_timezone_time=yes, tcl_cv_timezone_time=no)])
+ 	if test $tcl_cv_timezone_time = yes ; then
+ 	    AC_DEFINE(HAVE_TIMEZONE_VAR, 1, [Should we use the global timezone variable?])
+@@ -2452,17 +2452,17 @@ AC_DEFUN([TEA_BUGGY_STRTOD], [
+ 		    double value;
+ 		    value = strtod(infString, &term);
+ 		    if ((term != infString) && (term[-1] == 0)) {
+-			exit(1);
++			return 1;
+ 		    }
+ 		    value = strtod(nanString, &term);
+ 		    if ((term != nanString) && (term[-1] == 0)) {
+-			exit(1);
++			return 1;
+ 		    }
+ 		    value = strtod(spaceString, &term);
+ 		    if (term == (spaceString+1)) {
+-			exit(1);
++			return 1;
+ 		    }
+-		    exit(0);
++		    return 0;
+ 		}], tcl_cv_strtod_buggy=ok, tcl_cv_strtod_buggy=buggy,
+ 		    tcl_cv_strtod_buggy=buggy)])
+ 	if test "$tcl_cv_strtod_buggy" = buggy; then

--- a/repos/spack_repo/builtin/packages/expect/package.py
+++ b/repos/spack_repo/builtin/packages/expect/package.py
@@ -38,6 +38,8 @@ class Expect(AutotoolsPackage):
 
     patch("xcode_12.patch", when="%apple-clang@12:")
     patch("expect_detect_tcl_private_header_os_x_mountain_lion.patch", when="@5.45:5.45.0")
+    # https://sourceforge.net/p/expect/patches/24/
+    patch("func_proto.patch", when="%gcc@13:")
 
     def configure_args(self):
         spec = self.spec

